cmd/kubectl-gadget/utils/trace.go

```diff
@@ -553,6 +553,15 @@ func waitForTraceState(traceID string, expectedState string) (*gadgetv1alpha1.Tr
 	return waitForCondition(traceID, func(trace *gadgetv1alpha1.Trace) bool { return trace.Status.State == expectedState })
 }
 
+// waitForNoOperation waits for the traces with the ID received as parameter to
+// not have an operation.
+func waitForNoOperation(traceID string) (*gadgetv1alpha1.TraceList, error) {
+	return waitForCondition(traceID, func(trace *gadgetv1alpha1.Trace) bool {
+		_, present := trace.ObjectMeta.Annotations[GADGET_OPERATION]
```

Comment:
Should it check that `trace.ObjectMeta.Annotations` is not nil before?

Reply:
I added this even thought I am not sure of the returned value particularly because this should never happen:
[
```golang
	// Remove annotations first to avoid another execution in the next
	// reconciliation loop.
	withAnnotation := trace.DeepCopy()
	annotations := trace.GetAnnotations()
	delete(annotations, GADGET_OPERATION)
	for k := range params {
		delete(annotations, GADGET_OPERATION+"-"+k)
	}
	trace.SetAnnotations(annotations)
```](https://github.com/kinvolk/inspektor-gadget/blob/4efa1a7b39bf6bae1a183a96003f8b0c9d3fb824/pkg/controllers/trace_controller.go#L224-L232)

They would be empty but never `nil`.